src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -46,6 +46,8 @@ internal class AnalyzerExecutor
         private readonly Func<DiagnosticAnalyzer, bool> _shouldSkipAnalysisOnGeneratedCode;
         private readonly Func<Diagnostic, DiagnosticAnalyzer, Compilation, CancellationToken, bool> _shouldSuppressGeneratedCodeDiagnostic;
         private readonly Func<SyntaxTree, TextSpan, bool> _isGeneratedCodeLocation;
+        private readonly Func<DiagnosticAnalyzer, SyntaxTree, bool> _isAnalyzerSuppressedForTree;
```

Comment:
Using a delegate instead of performing a method call is going to introduce significant overhead. Is there a reason why we need this indirection?

Reply:
This type is created once per driver instance, which is once per compilation, so it shouldn't be that expensive. I have kept the consistency with other similar fields in the file. If we find that these allocations show up in performance traces, we should cleanup this whole file to avoid delegates.

Reply:
I never said the allocation is expensive, I said the *invocation* is expensive. Calling through a delegate is more expensive than calling a method.